lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1279,8 +1279,10 @@ sub progress_info ($self) {
 
 sub account_result_size ($self, $result_name, $size) {
     my $job_id = $self->id;
-    log_trace("Accounting size of $result_name for job $job_id: $size");
-    $self->update({result_size => \"coalesce(result_size, 0) + $size"});
+    my $dbh = $self->result_source->schema->storage->dbh;
+    log_trace "Accounting size of $result_name for job $job_id: $size";
+    # update via raw query to avoid exception in case the job has already been deleted meanwhile (see poo#119866)
+    $dbh->prepare('UPDATE jobs SET result_size = coalesce(result_size, 0) + ? WHERE id = ?')->execute($size, $job_id);
```

Comment:
And i would probably make it boolean if that's all the tests need.

Reply:
Why are you using DBI directly here?
I thought wrapping this into an eval and checking the eval error for "row not found" would be enough...
If we don't check the DBI error at all then we might miss other errors.

Reply:
Are we not using `RaiseError` here?

Reply:
Ah, I see

Reply:
> If we don't check the DBI error at all then we might miss other errors.

I don't think so. I've actually tested a syntax error or a wrong table/column name and in these kinds of errors still throw an exception.

> I thought wrapping this into an eval and checking the eval error for "row not found" would be enough

We'd also needed to check for yet another error message because when `discard_changes` has been called before the error message changes. Not sure whether that's the only case. I would say checking possibly localized and changing error messages via regex is the worst option we should only take if there's no other option.

Reply:
I actually like them as well so, updated the change.

Reply:
sorry @kraih :)
my comment was supposed to be a seperate discussion...